### PR TITLE
migrate unit tests to mocha. 

### PR DESCRIPTION
**Pull request by DonutEspresso**
Take 2, now with CHANGELOG.md.

Still have to figure out what to do with CORS. Wondering if @leitsubomi's audit log plugin should be merged into this repo instead as part of 5.0?

### CODE REVIEW

File: lib/plugins/jsonBodyParser.js [hunk 1]
```diff
@@ -20,40 +20,48 @@ var bodyReader = require('./bodyReader');
...
                 e.message));
-            return;
         }
 
-        if (options.mapParams !== false) {
+        if (opts.mapParams === true) {
             if (Array.isArray(params)) {
+                // if req.params exists, we have url params. we can't map an
+                // array safely onto req.params, throw an error.
+                if (req.params &&
```
Comment (micahr) [inline lib/plugins/jsonBodyParser.js:45]: Is this new code, or just new since you moved it over?

File: lib/pre/pause.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 // Copyright 2012 Mark Cavage, Inc.  All rights reserved.
 
+/* jscs:disable requireCamelCaseOrUpperCaseIdentifiers */
```
Comment (micahr) [inline lib/pre/pause.js]: pleasing our linting overlords

File: lib/utils/httpDate.js [hunk 3]
```diff
@@ -11,9 +11,12 @@
  * @returns  {String}     formatted dated object
  */
 module.exports = function httpDate(now) {
```
Comment (micahr) [inline lib/utils/httpDate.js:13]: Should probably call this out as an API change. `now` will no longer be set to the current `Date` if something a variable is passed in that is falsey.

File: test/fieldedTextParser.test.js [hunk 4]
```diff
@@ -1,136 +1,125 @@
-/**
- * Module dependencies
- */
-
 'use strict';
 
-if (require.cache[__dirname + '/lib/helper.js']) {
-    delete require.cache[__dirname + '/lib/helper.js'];
```
Comment (micahr) [inline test/fieldedTextParser.test.js:8]: yikes.

**Comment by micahr** [comment]
Looks great, it would also be good to break out some of the documentation from the main repo. But that should be a separate issue entirely.

File: lib/plugins/jsonBodyParser.js (hunk 1 above)
Comment (DonutEspresso) [inline lib/plugins/jsonBodyParser.js:45]: Yeah - req.params defaults to false now, so this logic handles some of the weird use cases and fixes existing issues where mapParams stomps over url variables.

File: lib/utils/httpDate.js (hunk 3 above)
Comment (DonutEspresso) [inline lib/utils/httpDate.js:13]: Yikes, didn't even thinking about that. httpDate isn't exposed to the public and is internal only - doesn't look like the falsy case here is being exercised so I think I will just remove it.

**Comment by micahr** [comment]
Can this be merged yet?

**Comment by micahr** [comment]
going to merge to so I can write tests in Mocha :)

**Pull request merged**